test/e2e/common/node/runtimeclass.go

```diff
@@ -375,12 +386,19 @@ func createRuntimeClass(ctx context.Context, f *framework.Framework, name, handl
 	return rc.GetName()
 }
 
-func expectPodRejection(ctx context.Context, f *framework.Framework, pod *v1.Pod) {
-	_, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(ctx, pod, metav1.CreateOptions{})
-	gomega.Expect(err).To(gomega.HaveOccurred(), "should be forbidden")
+// expectPodRejection is testing that kubelet cannot admit a runtime class pod
+// rejections happen in the admission and we expect a forbidden error for these cases.
+func expectPodRejection(ctx context.Context, f *framework.Framework, rcName string) error {
+	pod := e2eruntimeclass.NewRuntimeClassPod(rcName)
+	_, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(ctx, pod, metav1.CreateOptions{DryRun: []string{metav1.DryRunAll}})
+	if err == nil {
+		return fmt.Errorf("should get forbidden error")
+	}
 	if !apierrors.IsForbidden(err) {
-		framework.Failf("expected forbidden error, got %#v", err)
+		return err
 	}
+	// in this case means we got the forbidden error
+	return err
```

Comment:
Sort of weird to return an error here. Either call this method something like "dryRunRuntimeClassPod", and just have the caller check the error, or keep it "expectPodRejection" and return nil in this case (and only this case).

Reply:
updated.